readthedocs/search/utils.py

```diff
@@ -153,24 +155,27 @@ def _indexing_helper(html_objs_qs, wipe=False):
                 delete_objects_in_es.delay(**kwargs)
 
 
-def _remove_newlines_from_dict(highlight):
-    """
-    Recursively change results to turn newlines into periods.
+def _get_inner_hits_highlights(hit, logging=False):
+    """Returns highlight dict and does conditional logging of the same."""
+    if hasattr(hit, 'highlight'):
+        highlight_dict = hit.highlight.to_dict()
 
-    See: https://github.com/rtfd/readthedocs.org/issues/5168
-    :param highlight: highlight dict whose contents are to be edited.
-    :type highlight: dict
-    :returns: dict with all the newlines changed to periods.
-    :rtype: dict
-    """
-    for k, v in highlight.items():
-        if isinstance(v, dict):
-            highlight[k] = _remove_newlines_from_dict(v)
-        else:
-            # elastic returns the contents of the
-            # highlighted field in a list.
-            if isinstance(v, list):
-                v_new_list = [res.replace('\n', '. ') for res in v]
-                highlight[k] = v_new_list
-
-    return highlight
+        if logging:
+            log.debug('API Search highlight: %s', pformat(highlight_dict))
+
+        return highlight_dict
+    return {}
+
+
+def _get_sorted_results(results, source_key='_source', logging=False):
+    """Sort results according to their score and return a generator expression."""
+    sorted_results = (
```

Comment:
Do we really need this to be a generator? What is the goal of it versus just using a list?  It looks like we're just calling `list` on it both times we use it, which means it should probably just return a list :)

Reply:
I thought that returning a generator expression would be efficient than returning a huge list of results?